### PR TITLE
The filename filter is now matching on callsite instead of implemsite

### LINKED ISSUE #143: Match on the file where the function is called, and not the one where it's defined
Currently, the `.filename` filter is matching on the file where the function is defined. It might be cool to match on the file where the function is __called__ instead, to tell apart call sites, instead of homonym functions.

Thoughs?

### PR DESCRIPTION
**Pull request by kkadosh**
Currently, the `.filename` and `.filename_r` filters are matching in the file where the function is __defined__, and not where the function is __called__. This pull-request inverts this behaviour.

This should close #143

### CODE REVIEW

File: src/sp_disabled_functions.c [hunk 1]
```diff
@@ -213,12 +213,49 @@ static bool is_param_matching(zend_execute_data* execute_data,
   return false;
 }
 
+static zend_execute_data* is_file_matching(
+    zend_execute_data* const execute_data,
+    sp_disabled_function const* const config_node,
+    char const* const current_filename) {
+#define ITERATE(ex)                                            \
```
Comment (buixor) [inline src/sp_disabled_functions.c]: better check that ex != NULL before `ex = ex->prev`, smells null deref in weird context (ie. ob_start etc)

Comment (jvoisin) [reply src/sp_disabled_functions.c]: This check should be added right at the beginning on  `should_disable` in my opinion.

**Comment by blotus** [comment]
I'm not sure that iterating like this on execute_data will work in all cases (for example, what about functions that take a callback as a parameter ?)

**Comment by jvoisin** [comment]
I don't think that this is in the scope of this PR. What would be the intended behaviour anyway? Technically, the function isn't used in any file :/

**Pull request merged**